runtime/cudaq/platform/default/rest_server/helpers/RestRemoteClient.cpp

```diff
@@ -218,6 +254,304 @@ class RemoteRestRuntimeClient : public cudaq::RemoteRuntimeClient {
+        cudaq::utils::unzip(resultFilePath, unzipDir);
+        std::filesystem::path resultJsonFile =
+            unzipDir / (reqId + "_result.json");
+        if (!std::filesystem::exists(resultJsonFile)) {
+          if (optionalErrorMsg)
+            *optionalErrorMsg =
+                "Unexpected response file: missing the result JSON file.";
+          return false;
+        }
+        std::ifstream t(resultJsonFile.string());
+        std::string resultJsonFromFile((std::istreambuf_iterator<char>(t)),
+                                       std::istreambuf_iterator<char>());
+        try {
+          resultJs["response"] = json::parse(resultJsonFromFile);
+        } catch (...) {
+          if (optionalErrorMsg)
+            *optionalErrorMsg = "Failed to parse the response JSON from file.";
+          return false;
+        }
+        cudaq::info(
+            "Delete response zip file {} and its inflated contents in {}",
+            resultFilePath.c_str(), unzipDir.c_str());
+        std::filesystem::remove(resultFilePath);
+        std::filesystem::remove_all(unzipDir);
+      }
+
+      if (!resultJs.contains("response")) {
+        if (optionalErrorMsg)
+          *optionalErrorMsg = "Unexpected response from the NVCF invocation. "
+                              "Missing the 'response' field.";
+        return false;
+      }
+      if (!resultJs["response"].contains("executionContext")) {
+        if (optionalErrorMsg)
+          *optionalErrorMsg = "Unexpected response from the NVCF response. "
+                              "Missing the required field 'executionContext'.";
+        return false;
+      }
+      resultJs["response"]["executionContext"].get_to(io_context);
+      return true;
+    } catch (std::exception &e) {
+      if (optionalErrorMsg)
+        *optionalErrorMsg = e.what();
+      return false;
+    }
+  }
+
+  // Upload a job request as an NVCF asset.
+  // Return asset Id on success. Otherwise, return null.
+  std::optional<std::string>
+  uploadRequest(const cudaq::RestRequest &jobRequest) {
+    json requestJson;
+    requestJson["contentType"] = "application/json";
+    requestJson["description"] = "cudaq-nvcf-job";
+    try {
+      auto headers = getHeaders();
+      auto resultJs =
+          m_restClient.post(nvcfAssetUrl(), "", requestJson, headers, false);
+      const std::string uploadUrl = resultJs["uploadUrl"];
+      const std::string assetId = resultJs["assetId"];
+      cudaq::info("Upload NVCF Asset Id {} to {}", assetId, uploadUrl);
+      std::map<std::string, std::string> uploadHeader;
+      // This must match the request to create the upload link
+      uploadHeader["Content-Type"] = "application/json";
+      uploadHeader["x-amz-meta-nvcf-asset-description"] = "cudaq-nvcf-job";
+      json jobRequestJs = jobRequest;
+      m_restClient.put(uploadUrl, "", jobRequestJs, uploadHeader, false);
+      return assetId;
+    } catch (...) {
+      return {};
+    }
```

Comment:
Indeed, the caller will log an error message: https://github.com/1tnguyen/cuda-quantum/blob/791d2134eff41d4ffbece89d72fafeaa75aeba06/runtime/cudaq/platform/default/rest_server/helpers/RestRemoteClient.cpp#L417-L420

My thinking was that we just report the overall failure of this upload operation rather than letting the individual `post` and `put` throw.